src/cirrus/lib/statedb.py

```diff
@@ -366,21 +382,18 @@ def set_processing(
         now = isotimestamp if isotimestamp else datetime.now(UTC).isoformat()
         key = self.payload_id_to_key(payload_id)
 
-        expr = (
-            "SET "
-            "created = if_not_exists(created, :created), "
-            "state_updated=:state_updated, updated=:updated, "
-            "executions = list_append(if_not_exists(executions, :empty_list), :exes)"
-        )
+        expr = "SET state_updated=:state_updated, updated=:updated"
         return self.table.update_item(
             Key=key,
             UpdateExpression=expr,
+            ConditionExpression=(
+                "begins_with(state_updated, :claim) " "and contains(executions, :exec)"
+            ),
```

Comment:
I'm not sure you need to check for the execution ARN? If the state is `CLAIMED` then I _think_ we have an invariant here where executions must alread contain the execution ARN. Maybe no harm in checking I guess? I'm not sure.

I do think if you are performing two state updates in `ProcessPayload.__call__()` that could conditionally fail you need to ensure each is explicitly wrapped in an exception handler so you can differentiate between `ConditionalCheckFailedException` for each of them. But I think this is the comment you already have in that handler so I'll stop talking now.

Reply:
I _agree_ that this is redundant, and will remove.  Otherwise, this PR is ready for your more thorough review.